packages/framework/aqueduct/src/data-objects/pureDataObject.ts

```diff
@@ -24,13 +24,25 @@ import { handleFromLegacyUri } from "@fluidframework/request-handler";
 import { serviceRoutePathRoot } from "../container-services";
 import { defaultFluidObjectRequestHandler } from "../request-handlers";
 
-// eslint-disable-next-line @typescript-eslint/ban-types
-export interface IDataObjectProps<O = object, S = undefined> {
+export interface DataObjectTypes {
```

Comment:
this is the real meat of the change. here we defined the mapped type that consumer use to specify their generic arguments

Reply:
Love this change!

Would be nice to put friendlier names on these, maybe something like `OptionalProvider`, `InitialState`, `EmittedEvents` or something?